python/pypto/jit/decorator.py

```
@@ -262,6 +405,159 @@ def _get_pl_dtype_map() -> dict[str, Any]:
+def _func_name_lookup(func: Any) -> dict[str, Any]:
+    """Return ``func.__globals__`` merged with closure free-var bindings.
+
+    A function defined inside a test method (or any other enclosing scope)
+    captures module-level helpers (``HIDDEN``, ``ROWS``, ...) as closure free
+    vars, not as globals — both namespaces have to be inspected for static
+    shape-element resolution to find them.
+    """
+    out: dict[str, Any] = dict(getattr(func, "__globals__", {}))
+    co_freevars = getattr(getattr(func, "__code__", None), "co_freevars", ())
+    closure = getattr(func, "__closure__", None) or ()
+    for fv_name, cell in zip(co_freevars, closure):
```

Ruff (0.15.14)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)